spec/api.yaml

```diff
@@ -0,0 +1,269 @@
+      operationId: listTables
+      tags:
+        - Table API
+      responses:
+        '200':
+          description: A list of Tables
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ListTablesResponse"
+        '500':
+          description: An unexpected error occurred
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+  /table/{table_name}/info:
+    get:
+      summary: Get a Table
+      description: Returns the information about the Table identified by name.
+      operationId: getTable
+      tags:
+        - Table API
+      parameters:
+        - name: table_name
+          in: path
+          description: Unique Table name, unique to this implementation
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: A Table
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Table"
+        '404':
+          description: The table doesn't exist
+        '500':
+          description: An unexpected error occurred
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+  /table/{table_name}/data:
+    get:
+      summary: Fetch data from a Table
+      description: Optional operation that returns data from a Table
+      operationId: getData
+      tags:
+        - Table API
+      parameters:
+        - name: table_name
+          in: path
+          description: Unique Table name, unique to this implementation
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Data from the Table
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/TableData"
+        '404':
+          description: The table doesn't exist
+        '500':
+          description: An unexpected error occurred
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+  /search:
+    post:
+      summary: Perform a search on Tables
+      description: Optional operation that accepts a SearchRequest and returns a TableData
+      operationId: search
+      requestBody:
+        description: Query to execute
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SearchRequest"
+      tags:
+        - Search API
+      responses:
+        '200':
+          description: Query results returned as TableData
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/TableData"
+        '400':
+          description: The query is invalid
```

Comment:
Should we change this to "Error in request headers or body, for example if the query is invalid. Details are provided in the ErrorResponse body."?

Reply:
Makes sense!